Prometheus alerts doc

ppatierno opened the pull request:
### Type of change

- Documentation

### Description

This is the documentation PR for the Prometheus alerts support.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

ppatierno commented:
Do you mean the alerts in general so something like the Prometheus Alerting documentation or describing the alert examples we have included?

scholzj commented:
Yeah ... I mean about describing the alerts which are in the example file for example. Or even Prometheus alerts in general, but separate it form the Prometheus deployment.

ppatierno commented:
I have renamed the "Alerting" section in "Prometheus alert manager" for describing what we provide and the resources to deploy for that. Under this section, I added a subsection with alert examples description.
The Prometheus alerts description in general should be a different chapter at the same level of the main ones, right?

scholzj commented:
Yeah ... at least my 2¢ ... I would leave in the appendix the part about installing Prometheus, Grafana and Prometheus Alerting it self. At least until we use the Prometheus Operator for that. But I would move the chapters about our dashboards and our alerts to regular chapters in the main 
 documentation. (Of course moving the Grafana stuff is probably a separate PR since this is about alerts).

I do not know what do @laidan6000 and @ncbaratta think about this.

ncbaratta commented on documentation/book/appendix_metrics.adoc:
```diff
@@ -50,18 +50,31 @@ The following resources are defined:
 * A `ConfigMap` to manage the configuration of Prometheus Server.
 * A `Service` to provide an easy to reference hostname for other services to connect to Prometheus server (such as Grafana).
 
+Prometheus also provides an alerting system through the https://prometheus.io/docs/alerting/alertmanager/[alert manager] component.
```
```suggestion
Prometheus also provides an alerting system through the link:https://prometheus.io/docs/alerting/alertmanager/[alert manager] component.
```

ncbaratta commented on documentation/book/appendix_metrics.adoc:
```diff
@@ -50,18 +50,31 @@ The following resources are defined:
 * A `Service` to provide an easy to reference hostname for other services to connect to Prometheus server (such as Grafana).
 
+Prometheus also provides an alerting system through the https://prometheus.io/docs/alerting/alertmanager/[alert manager] component.
+In order to enable alerting, the provided https://raw.githubusercontent.com/strimzi/strimzi-kafka-operator/{GithubVersion}/metrics/examples/prometheus/alerting-rules.yaml[`alerting-rules.yaml`] YAML file describes a `ConfigMap` resource which defines some sample alerting rules on Kafka and Zookeeper metrics.
```
Will this work up and downstream? With the GithubVersion attribute?

ppatierno replied:
I don't think so but it should e part of a different work of rewording this section for upstream.

ncbaratta commented on documentation/book/appendix_metrics.adoc:
```diff
@@ -50,18 +50,31 @@ The following resources are defined:
 * A `Service` to provide an easy to reference hostname for other services to connect to Prometheus server (such as Grafana).
 
+Prometheus also provides an alerting system through the https://prometheus.io/docs/alerting/alertmanager/[alert manager] component.
+In order to enable alerting, the provided https://raw.githubusercontent.com/strimzi/strimzi-kafka-operator/{GithubVersion}/metrics/examples/prometheus/alerting-rules.yaml[`alerting-rules.yaml`] YAML file describes a `ConfigMap` resource which defines some sample alerting rules on Kafka and Zookeeper metrics.
```
```suggestion
In order to enable alerting, the provided link:https://raw.githubusercontent.com/strimzi/strimzi-kafka-operator/{GithubVersion}/metrics/examples/prometheus/alerting-rules.yaml[`alerting-rules.yaml`] file describes a `ConfigMap` resource which defines sample alerting rules on Kafka and Zookeeper metrics.
```

ncbaratta commented:
@scholzj @ppatierno I'm building the guide locally so I can see where things fall and then can answer your question.

ncbaratta commented:
Okay, for now, I say leave it in the appendix, but I think we're going to have to a full architecture review sometime soon to make sure we keep things in logical places. There are several locations to find information about Prometheus right now and I can see that being annoying to someone looking for the right info about Prometheus.

ppatierno commented:
@scholzj @ncbaratta any more comments on this one so that we can go forward? I think that the rewording for having it downstream should be part of a different work and related PR.

ncbaratta commented:
Nope, I'm good.

scholzj commented:
I'm good @ppatierno ... Thanks
